consensus/src/block_storage/block_store.rs

```diff
@@ -96,6 +96,40 @@ pub struct BlockStore {
     time_service: Arc<dyn TimeService>,
 }
 
+pub fn update_counters_and_prune_blocks(
+    block_tree: Arc<RwLock<BlockTree>>,
+    storage: Arc<dyn PersistentLivenessStorage>,
+    root: Arc<ExecutedBlock>,
+    blocks_to_commit: &[Arc<ExecutedBlock>],
+    block_to_commit: Arc<ExecutedBlock>,
+) {
+    update_counters_for_committed_blocks(blocks_to_commit);
```

Comment:
hmm, this is not right, this expects the real executed blocks (it reports txn status, last version etc) but we pass in ordered blocks.

need to think how to make this work and clean

Reply:
add parameters to the closure?